eng/pipelines/templates/stages/source-build-stages.yml

```diff
@@ -0,0 +1,242 @@
+parameters:
+- name: isBuiltFromVmr
+  type: boolean
+  default: false
+
+- name: pool_Linux
+  type: object
+
+- name: pool_LinuxArm64
+  type: object
+
+# Scope of jobs which are executed
+- name: scope
+  type: string
+
+- name: useMicrosoftBuildAssetsForTests
+  type: boolean
+
+- name: legs
+  type: object
+
+stages:
+- stage: VMR_SourceOnly_Build
+  displayName: VMR Source-Only Build
+  dependsOn: []
+  variables:
+  - template: ../variables/vmr-build.yml
+    parameters:
+      vmrBranch: ${{ variables.VmrBranch }}
+      buildSourceOnly: true
+  jobs:
+  - ${{ each leg in parameters.legs }}:
+    - template: ../jobs/vmr-build.yml
+      parameters:
+        # CentOS
+        ${{ if eq(leg.distro, 'centos') }}:
+          buildName: ${{ format(leg.buildNameFormat, variables.centOSStreamName) }}
+          container:
```

Comment:
What do you think about declaring parameters (object) for the supported distro containers?  This would help reduce the duplication across build/test.

Reply:
Similar to my other [response](https://github.com/dotnet/sdk/pull/48703#discussion_r2064451156), in order to define a value to pass as input, you'd need to reference variables, which you can't do in the calling template, source-build.yml.

Reply:
Correct. For default values of parameters, you can't use template expressions which would be required in order to reference the variable.